### PR TITLE
Refactor `log_reports.py` for Improved Logging, File Processing, and Slack Payload Handling

### PR DESCRIPTION
**Pull request by Mefisto04**
This PR introduces several improvements to the `log_reports.py` script, including:

1. **Logging Enhancements:**
   - Added a context-managed `setup_logging` function to configure logging with a more detailed format for better debugging.

2. **File Processing Refactor:**
   - Created a new `process_log_file` function to encapsulate log file processing, reducing redundancy and improving error handling.
   - Enhanced error handling to include specific exceptions like `FileNotFoundError` and `JSONDecodeError` for more precise logging.

3. **Slack Payload Handling Improvements:**
   - Refactored the Slack message preparation to improve readability and structure.
   - Included a fallback message when the total message length exceeds the maximum allowed.

4. **Code Cleanup:**
   - Improved code readability by separating concerns and adding comments where necessary.
   - Ensured consistent log file cleanup with error handling for potential `OSError` cases.

These changes aim to make the script more maintainable, enhance logging detail, and improve the reliability of Slack notifications.

### CODE REVIEW

**Comment by Mefisto04** [comment]
hey @qgallouedec, i have make a pr for this issue #2249, please review all the changes that i have made.

**Comment by Mefisto04** [comment]
hey @qgallouedec , i have made a pr for this issue #2237  , please review all the changes that i have made.

File: scripts/log_reports.py [hunk 1]
```diff
@@ -20,144 +20,135 @@
 
 from tabulate import tabulate
 
-
-MAX_LEN_MESSAGE = 2900  # slack endpoint has a limit of 3001 characters
+MAX_LEN_MESSAGE = 2900  # Slack endpoint limit of 3001 characters
```
Comment (qgallouedec) [inline scripts/log_reports.py:23]:
```suggestion
MAX_LEN_MESSAGE = 2900  # Slack endpoint has limit of 3001 characters
```

File: scripts/log_reports.py [hunk 2]
```diff
@@ -20,144 +20,135 @@
...
-
-def main(slack_channel_name=None):
-    failed = []
-    passed = []
-
+# Set up logging using context manager
+def setup_logging():
+    logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+setup_logging()
```
Comment (qgallouedec) [inline scripts/log_reports.py:32]: Why do you need it in a function?

File: scripts/log_reports.py [hunk 3]
```diff
@@ -20,144 +20,135 @@
...
         print(f"### {message}")
-    else:
-        payload.append(no_error_payload)
-
-    if os.environ.get("TEST_TYPE", "") != "":
-        from slack_sdk import WebClient
 
         if len(message) > MAX_LEN_MESSAGE:
-            message = f"There are {total_num_failed} failed tests in total ! Cannot display the entire summary - please check the action results directly"
+            message = f"There are {total_num_failed} failed tests in total! Please check the action results directly."
```
Comment (qgallouedec) [inline scripts/log_reports.py:115]:
```suggestion
            message = f"❌ There are {total_num_failed} failed tests in total! Please check the action results directly."
```

File: scripts/log_reports.py [hunk 4]
```diff
@@ -20,144 +20,135 @@
...
-        client.chat_postMessage(channel=f"#{slack_channel_name}", text=message, blocks=payload)
+        slack_client = WebClient(token=os.environ.get("SLACK_API_TOKEN"))
+        slack_client.chat_postMessage(channel=f"#{slack_channel_name}", text=message, blocks=payload)
 
+    else:
+        payload.append({
+            "type": "section",
+            "text": {
+                "type": "plain_text",
+                "text": "🌞 No failures! All tests passed successfully.",
```
Comment (qgallouedec) [inline scripts/log_reports.py:147]:
```suggestion
                "text": "✅ No failures! All tests passed successfully.",
```

**Comment by qgallouedec** [comment]
Thanks for helping improving this @Mefisto04. Can you make sure to run `make precommit`? Otherwise it looks good to me!

**Comment by Mefisto04** [comment]
hey @qgallouedec i have commits all the changes that you have provided, please review this

File: scripts/log_reports.py (hunk 2 above)
Comment (qgallouedec) [reply scripts/log_reports.py:32]: `log_report` is a script, so its code is not intended to be imported or reused. And there's no maintenance involved, it's just a line called once. Having a comment will do just fine.

```suggestion
# Setup logging
logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
```

**Review by qgallouedec — APPROVED** [review]
Thanks again @Mefisto04

File: scripts/log_reports.py [hunk 5]
```diff
@@ -21,142 +21,146 @@
...
 
-MAX_LEN_MESSAGE = 2900  # slack endpoint has a limit of 3001 characters
+MAX_LEN_MESSAGE = 2900  # Slack endpoint has a limit of 3001 characters
 
 parser = argparse.ArgumentParser()
 parser.add_argument("--slack_channel_name", default="trl-push-ci")
 
-# Set up logging
-logging.basicConfig(level=logging.INFO)
+# Setup logging
```
Comment (qgallouedec) [inline scripts/log_reports.py:29]:
```suggestion
# Set up logging
```

**Pull request merged**